trieste/acquisition/function/continuous_thompson_sampling.py

```diff
@@ -149,12 +174,14 @@ def update_acquisition_function(
 
         if new_function is not self._trajectory:  # need to negate again if not modified in place
             self._trajectory = new_function
-            self._negated_trajectory = negate_trajectory_function(new_function)
+            self._negated_trajectory = negate_trajectory_function(new_function, self._select_output)
 
         return self._negated_trajectory
 
 
-def negate_trajectory_function(function: TrajectoryFunction) -> TrajectoryFunction:
+def negate_trajectory_function(
+    function: TrajectoryFunction, select_output: Optional[Callable[[TensorType], TensorType]] = None
+) -> TrajectoryFunction:
     """
     Return the negative of trajectories so that our acquisition optimizers (which are
     all maximizers) can be used to extract the minimizers of trajectories.
```

Comment:
The logic around negate_trajectory_function is getting increasingly ugly, but I can't immediately think of a better approach.

In the meantime, maybe update the docstring (speciically the next paragraph) as this function now doesn't just negate the call method, but also applies the output selector.